### PR TITLE
RDoc-927 implementation

### PR DESCRIPTION
**Pull request by myarichuk**
Add relevant article and code sample on how replication can be configured in Embedded RavenDB

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿# Replication : Using Embedded instance
+
+## Overview
+[Replication](../../../server/clustering/replication/replication) works by using long-living TCP connections between cluster nodes, or in case of [External Replication](../../../server/ongoing-tasks/external-replication) instances external to the cluster. 
+Esentially, Embedded RavenDB is the same as non-embedded, their only difference is what process is a host. [Here](../../../server/embedded) you can read more in-depth about Embedded RavenDB functionality.
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown]:
```suggestion
Essentially, Embedded RavenDB is the same as non-embedded, their only difference is what process is a host. [Here](../../../server/embedded) you can read more in-depth about Embedded RavenDB functionality.
```

File: Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Configuring replication between embedded instance and a cluster
+One possibility would be to configure it through [the studio](../../../studio/server/cluster/add-node-to-cluster#add-another-node-to-the-cluster), by adding the Embedded instance to the cluster, then adding the database replicated to relevant [database group](../../../server/clustering/distribution/distributed-database).  
+Studio of the Embedde instance can be opened in the following way:
+{CODE Embedded_OpenStudio@Server\ReplicationEmbedded.cs /}
+
+Another possibility would be to configure it programmatically.
+
+{CODE Embedded_Replication_Setup@Server\ReplicationEmbedded.cs /}
+
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown]: Need to mention that by default, embedded is using random port, and if you want to replace _to_ embedded instance, you need to fix the port

File: Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿# Replication : Using Embedded instance
+
+## Overview
+[Replication](../../../server/clustering/replication/replication) works by using long-living TCP connections between cluster nodes, or in case of [External Replication](../../../server/ongoing-tasks/external-replication) instances external to the cluster. 
+Esentially, Embedded RavenDB is the same as non-embedded, their only difference is what process is a host. [Here](../../../server/embedded) you can read more in-depth about Embedded RavenDB functionality.
+
+## Configuring replication between embedded instance and a cluster
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown]: This is actually, adding embedded instance to cluster / database group, isn't it?

Comment (arekpalinski) [reply Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown]: I wonder if showing how to setup external replication using `UpdateExternalReplicationOperation` wouldn't be also good example here

Comment (myarichuk) [reply Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown]: I've added code to setup external replication + links to related studio articles

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/Raven.Documentation.Samples.csproj [hunk 4]
```diff
@@ -4,8 +4,9 @@
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="NodaTime" Version="2.4.2" />
-    <PackageReference Include="RavenDB.Client" Version="4.0.9-patch-40054" />
-    <PackageReference Include="RavenDB.TestDriver" Version="4.0.9-patch-40054" />
+    <PackageReference Include="RavenDB.Client" Version="4.1.3-nightly-20181108-0530" />
+    <PackageReference Include="RavenDB.Embedded" Version="4.1.3-nightly-20181108-0530" />
+    <PackageReference Include="RavenDB.TestDriver" Version="4.1.3-nightly-20181108-0530" />
```
[the remark this answers is not in the document]
Comment (myarichuk) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/Raven.Documentation.Samples.csproj]: moved.

File: Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿# Replication : Using Embedded instance
```
[the remark this answers is not in the document]
Comment (myarichuk) [reply Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/replication-and-embedded-instance.markdown]: moved.

File: Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/.docs.json [hunk 6]
```diff
@@ -52,5 +52,26 @@
         "4.1",
         "4.2"
     ]
-  }
+    },
+    
+    {
+        "Path": "replication-and-embedded-instance.markdown",
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/.docs.json]: @myarichuk you need to remove this I think?

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/Raven.Documentation.Samples.csproj (hunk 4 above)
[the remark this answers is not in the document]
Comment (ppekrol) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/Raven.Documentation.Samples.csproj]: @myarichuk can you remove this?

[the remark this answers is not in the document]
Comment (arekpalinski) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/Raven.Documentation.Samples.csproj]: I've reverted it

File: Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/.docs.json (hunk 6 above)
Comment (arekpalinski) [reply Documentation/4.0/Raven.Documentation.Pages/server/clustering/replication/.docs.json]: It points to 4.0 article which has "Running embedded RavenDB instances is supported starting from version 4.1."

**Pull request merged**